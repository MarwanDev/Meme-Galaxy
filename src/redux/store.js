import { configureStore } from "@reduxjs/toolkit";
import memesReducer from "./memesSlice/memesSlice";

const store = configureStore({
  reducer: {
    meme: memesReducer,
  },
});

export default store;
