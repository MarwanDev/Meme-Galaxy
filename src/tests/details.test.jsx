import { Details } from "../pages";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import renderer from "react-test-renderer";
import store from "../redux/store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

describe("Details", () => {
  it("renders each category correctly", () => {
    const details = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Details />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(details).toMatchSnapshot();
  });
  it("renders Category component correctly", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Details />
        </Provider>
      </BrowserRouter>
    );
  });
});
