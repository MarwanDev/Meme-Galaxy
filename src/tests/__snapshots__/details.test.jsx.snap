// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Details > renders each category correctly 1`] = `
[
  <nav
    className="navbar-container"
  >
    <a
      href="/"
      onClick={[Function]}
      style={
        {
          "backgroundColor": "#4369b2",
          "border": "none",
          "paddingTop": 3,
          "textDecoration": "none",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowBackIosNewIcon"
        focusable="false"
        style={
          {
            "backgroundColor": "#4369b2",
            "color": "#fff",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
        />
      </svg>
    </a>
    <h1
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#4369b2",
          "color": "#fff",
          "fontSize": 18,
          "margin": "auto",
          "textAlign": "center",
        }
      }
    >
      memes list
    </h1>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MicIcon"
      focusable="false"
      style={
        {
          "backgroundColor": "#4369b2",
          "color": "#fff",
          "marginLeft": "auto",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SettingsIcon"
      focusable="false"
      style={
        {
          "backgroundColor": "#4369b2",
          "color": "#fff",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
      />
    </svg>
  </nav>,
  <div
    className="details-header"
  >
    <h5
      style={
        {
          "backgroundColor": "#5288e2",
          "fontSize": 80,
          "textDecoration": "none",
        }
      }
    >
      ðŸ¤‘  
    </h5>
    <div
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#5288e2",
          "color": "#fff",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <p
        style={
          {
            "backgroundColor": "#5288e2",
            "fontSize": 20,
            "marginLeft": 12,
          }
        }
      >
        All
      </p>
      <p
        style={
          {
            "backgroundColor": "#5288e2",
          }
        }
      >
        0
         memes
      </p>
    </div>
  </div>,
  <div
    className="cards-container"
  />,
]
`;
